### PR TITLE
dev-build/make: Build & install html+pdf docs

### PR DESCRIPTION
**Pull request by mjsir911**
<details>
<summary>Newly added files</summary>

```
/usr/share/doc/make-4.4.1-r1/make.pdf
/usr/share/doc/make-4.4.1-r1/html/Windows-Jobserver.html
/usr/share/doc/make-4.4.1-r1/html/Choosing-the-Shell.html
/usr/share/doc/make-4.4.1-r1/html/Extending-make.html
/usr/share/doc/make-4.4.1-r1/html/Flavor-Function.html
/usr/share/doc/make-4.4.1-r1/html/Catalogue-of-Rules.html
/usr/share/doc/make-4.4.1-r1/html/Remaking-Makefiles.html
/usr/share/doc/make-4.4.1-r1/html/Guile-Integration.html
/usr/share/doc/make-4.4.1-r1/html/Execution.html
/usr/share/doc/make-4.4.1-r1/html/Guile-Interface.html
/usr/share/doc/make-4.4.1-r1/html/Search-Algorithm.html
/usr/share/doc/make-4.4.1-r1/html/Foreach-Function.html
/usr/share/doc/make-4.4.1-r1/html/make-Deduces.html
/usr/share/doc/make-4.4.1-r1/html/Conditional-Functions.html
/usr/share/doc/make-4.4.1-r1/html/Using-Variables.html
/usr/share/doc/make-4.4.1-r1/html/Options-Summary.html
/usr/share/doc/make-4.4.1-r1/html/Preparing.html
/usr/share/doc/make-4.4.1-r1/html/Loaded-Object-Example.html
/usr/share/doc/make-4.4.1-r1/html/Rule-Example.html
/usr/share/doc/make-4.4.1-r1/html/Pattern_002dspecific.html
/usr/share/doc/make-4.4.1-r1/html/Install-Command-Categories.html
/usr/share/doc/make-4.4.1-r1/html/Double_002dColon.html
/usr/share/doc/make-4.4.1-r1/html/Terminal-Output.html
/usr/share/doc/make-4.4.1-r1/html/Special-Targets.html
/usr/share/doc/make-4.4.1-r1/html/POSIX-Jobserver.html
/usr/share/doc/make-4.4.1-r1/html/Static-versus-Implicit.html
/usr/share/doc/make-4.4.1-r1/html/Flavors.html
/usr/share/doc/make-4.4.1-r1/html/Quick-Reference.html
/usr/share/doc/make-4.4.1-r1/html/Rule-Introduction.html
/usr/share/doc/make-4.4.1-r1/html/Wildcards.html
/usr/share/doc/make-4.4.1-r1/html/Implicit_002fSearch.html
/usr/share/doc/make-4.4.1-r1/html/How-Make-Works.html
/usr/share/doc/make-4.4.1-r1/html/Utilities-in-Makefiles.html
/usr/share/doc/make-4.4.1-r1/html/Make-Control-Functions.html
/usr/share/doc/make-4.4.1-r1/html/Implicit-Rule-Search.html
/usr/share/doc/make-4.4.1-r1/html/Canned-Recipes.html
/usr/share/doc/make-4.4.1-r1/html/Simple-Makefile.html
/usr/share/doc/make-4.4.1-r1/html/Functions.html
/usr/share/doc/make-4.4.1-r1/html/Parallel.html
```

</details>

### CODE REVIEW

File: dev-build/make/make-4.3-r1.ebuild [hunk 1]
```diff
@@ -64,9 +67,14 @@ src_configure() {
 	econf "${myeconfargs[@]}"
 }
 
+src_compile() {
+	emake all
```
Comment (thesamesam) [inline dev-build/make/make-4.3-r1.ebuild:71]: Let's just do the latest ebuild please, but for EAPI 8, you can do:
```suggestion
	emake all $(usev doc 'pdf html')
```

File: dev-build/make/make-4.4.1-r1.ebuild [hunk 2]
```diff
@@ -22,7 +22,7 @@ fi
 
 LICENSE="GPL-3+"
 SLOT="0"
-IUSE="guile nls static"
+IUSE="guile nls static doc"
```
Comment (thesamesam) [inline dev-build/make/make-4.4.1-r1.ebuild:25]: Sort.

File: dev-build/make/make-4.4.1-r1.ebuild [hunk 3]
```diff
@@ -32,8 +32,11 @@ RDEPEND="
 BDEPEND="
 	nls? ( sys-devel/gettext )
 	verify-sig? ( sec-keys/openpgp-keys-make )
+	doc? ( sys-apps/texinfo )
```
Comment (thesamesam) [inline dev-build/make/make-4.4.1-r1.ebuild:35]: Sort.

File: dev-build/make/make-9999.ebuild [hunk 4]
```diff
@@ -32,8 +32,11 @@ RDEPEND="
 BDEPEND="
 	nls? ( sys-devel/gettext )
 	verify-sig? ( sec-keys/openpgp-keys-make )
+	doc? ( sys-apps/texinfo )
 "
 
+DOCS="AUTHORS NEWS README*"
```
Comment (thesamesam) [inline dev-build/make/make-9999.ebuild:38]: Can we change to array please?

File: dev-build/make/make-9999.ebuild [hunk 5]
```diff
@@ -67,9 +70,14 @@ src_configure() {
...
 
+src_compile() {
+	emake all
+	use doc && make pdf html
+}
+
 src_install() {
-	emake DESTDIR="${D}" install
-	dodoc AUTHORS NEWS README*
+	use doc && HTML_DOCS="doc/make.html/." DOCS="$DOCS doc/make.pdf"
```
Comment (thesamesam) [inline dev-build/make/make-9999.ebuild:79]: Array, but also ${X} style.

File: dev-build/make/make-9999.ebuild (hunk 4 above)
Comment (mjsir911) [reply dev-build/make/make-9999.ebuild:38]: I haven't been able to successfully set DOCS=() with the wildcard README contained within it.

File: dev-build/make/make-9999.ebuild (hunk 5 above)
Comment (mjsir911) [reply dev-build/make/make-9999.ebuild]: done for HTML_DOCS but see above for DOCS

**Comment by thesamesam** [comment]
Rebase?

**Pull request closed without merging**